<a href="https://colab.research.google.com/github/senBenz/TP2/blob/main/Exercice1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
import numpy as np
l1=np.append(np.array(np.zeros(10)),np.ones(10))
print(l1)

[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]


Cette cellule crée un tableau NumPy de 20 éléments : d'abord 10 zéros, suivis de 10 uns concaténés.

In [13]:
integers=np.arange(10,51,1)
print(integers)

[10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33
 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50]


Ici, nous générons une séquence d'entiers allant de 10 à 50 (inclus).

In [14]:
even=np.arange(10,51,2)
print(even)

[10 12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42 44 46 48 50]


Cette commande génère uniquement les nombres pairs entre 10 et 50 en utilisant un pas de 2.

In [20]:
matrice=np.array([[1,0,0],[0,1,0],[0,0,1]])
print(matrice)

[[1 0 0]
 [0 1 0]
 [0 0 1]]


Ceci crée une matrice identité 3x3, où la diagonale principale est remplie de 1.

In [28]:
ligne1=np.arange(0,9,1)
even_matrice=np.array([ligne1]*3)
print(even_matrice)

[[0 1 2 3 4 5 6 7 8]
 [0 1 2 3 4 5 6 7 8]
 [0 1 2 3 4 5 6 7 8]]


Cette cellule crée une matrice en répétant trois fois une ligne contenant les chiffres de 0 à 8.

In [35]:
rnd = np.random.rand()
print(rnd)

0.6411202529287777


On génère ici un nombre décimal aléatoire compris entre 0 et 1.

In [39]:
echantillon = np.random.standard_normal(25)
print(echantillon)


[-1.35671354 -0.04128168 -0.32851486  2.01299504 -0.30250407  1.14582548
  1.18941772 -1.89609182  0.18995271 -1.10325905 -0.31087232  0.86406154
 -0.67787342  0.18788607  0.51594376 -0.81185935  2.27475759 -0.8390399
 -1.10789894 -1.8640502  -0.7466045   0.62539217  0.18005476  0.18728589
  0.31402848]


Ce bloc génère un échantillon de 25 nombres suivant une loi normale standard (moyenne 0, écart-type 1).

In [42]:
linear_spaced_values=np.arange(0,1,0.01)[:20]
print(linear_spaced_values)

[0.   0.01 0.02 0.03 0.04 0.05 0.06 0.07 0.08 0.09 0.1  0.11 0.12 0.13
 0.14 0.15 0.16 0.17 0.18 0.19]


Ici, on crée une séquence de 0 à 1 avec un pas de 0.01, puis on affiche les 20 premières valeurs.

In [55]:
matrice5by5=np.arange(1,26,1).reshape(5,5)
print(matrice5by5)

[[ 1  2  3  4  5]
 [ 6  7  8  9 10]
 [11 12 13 14 15]
 [16 17 18 19 20]
 [21 22 23 24 25]]


On crée un tableau de 1 à 25 que l'on reforme (reshape) en une matrice carrée de 5 lignes et 5 colonnes.

In [46]:
somme=np.sum(matrice5by5)
print(somme)

325


Calcule la somme totale de tous les éléments présents dans la matrice 5x5 (le résultat est 325).

In [47]:
ecarttype=np.std(matrice5by5)
print(ecarttype)

7.211102550927978


Calcule l'écart-type des valeurs de la matrice, ce qui mesure la dispersion des données.

In [70]:
column_sum=np.sum(matrice5by5,axis=0)
print(np.array(column_sum))


[55 60 65 70 75]


Calcule la somme de chaque colonne de la matrice en utilisant l'argument `axis=0`.